pkg/providers/vsphere/vmlifecycle/update_status_test.go

```diff
@@ -3596,12 +3590,11 @@ var _ = Describe("Snapshot status", func() {
 
 				Expect(vmCtx.VM.Status.RootSnapshots[0]).ToNot(BeNil())
 				Expect(vmCtx.VM.Status.RootSnapshots[0].Type).To(Equal(vmopv1.VirtualMachineSnapshotReferenceTypeUnmanaged))
-				Expect(vmCtx.VM.Status.RootSnapshots[0].Reference).To(BeNil())
+				// RootSnapshots[0] is unmanaged, so it doesn't have a corresponding snapshot name.
```

Comment:
The definition said for unmanaged snapshot, there's still a name? I guess the defination needs change?
```
                        Name is the name of the snapshot.
                        For Managed snapshots, this refers to the name of the VirtualMachineSnapshot resource.
                        For Unmanaged snapshots, this refers to the provider-specific snapshot name.
```